tests/e2e/e2e_test.go

```diff
@@ -88,6 +92,82 @@ func (s *IntegrationTestSuite) TestIBC() {
 	s.testIBCTokenTransfer()
 	s.testMultihopIBCTokenTransfer()
 	s.testFailedMultihopIBCTokenTransfer()
+
+	// stop hermes0 to prevent hermes0 relaying transactions
+	s.Require().NoError(s.dkrPool.Purge(s.hermesResource0))
+	HermesResource0Purged = true
+	s.testIBCBypassMsg()
+}
+
+func (s *IntegrationTestSuite) testIBCBypassMsg() {
+
+	// submit gov proposal to change bypass-msg param to
+	// ["/ibc.core.channel.v1.MsgRecvPacket",
+	//  "/ibc.core.channel.v1.MsgAcknowledgement",
+	//  "/ibc.core.client.v1.MsgUpdateClient"]
+	submitterAddr := s.chainA.validators[0].keyInfo.GetAddress()
+	submitter := submitterAddr.String()
+	proposalCounter++
+	s.govProposeNewBypassMsgs([]string{
+		sdk.MsgTypeURL(&ibcchanneltypes.MsgRecvPacket{}),
+		sdk.MsgTypeURL(&ibcchanneltypes.MsgAcknowledgement{}),
+		sdk.MsgTypeURL(&ibcclienttypes.MsgUpdateClient{})}, proposalCounter, submitter, standardFees.String())
+
+	// use hermes1 to test default ibc bypass-msg
+	//
+	// test 1: transaction only contains bypass-msgs, pass
+	s.T().Logf("testing transaction contains only ibc bypass messages")
+	ok := s.hermesTransfer(hermesCofigWithGasPrices, s.chainA.id, s.chainB.id, transferChannel, uatomDenom, 100, 1000, 1)
+	s.Require().True(ok)
+
+	scrRelayerBalanceBefore, dstRelayerBalanceBefore := s.queryRelayerWalletsBalances()
+
+	pass := s.hermesClearPacket(hermesCofigNoGasPrices, s.chainA.id, transferChannel)
+	s.Require().True(pass)
+	pendingPacketsExist := s.hermesPendingPackets(hermesCofigNoGasPrices, s.chainA.id, transferChannel)
+	s.Require().False(pendingPacketsExist)
+
+	// confirm relayer wallets do not pay fees
+	scrRelayerBalanceAfter, dstRelayerBalanceAfter := s.queryRelayerWalletsBalances()
+	s.Require().Equal(scrRelayerBalanceBefore.String(), scrRelayerBalanceAfter.String())
+	s.Require().Equal(dstRelayerBalanceBefore.String(), dstRelayerBalanceAfter.String())
+
+	// test 2: test transactions contains both bypass and non-bypass msgs (ibc msg mix with time out msg)
+	s.T().Logf("testing transaction contains both bypass and non-bypass messages")
+	ok = s.hermesTransfer(hermesCofigWithGasPrices, s.chainA.id, s.chainB.id, transferChannel, uatomDenom, 100, 1, 1)
+	s.Require().True(ok)
+	// make sure that the transaction is timeout
+	time.Sleep(3)
```

Comment:
Can u expand on why this is required?

Reply:
in code line 137, `hermesTransfer` with ` --timeout-height-offset` = 1,  
so here the sleep is just to make sure that the when hermes clear the transaction, the transaction is timeout. 

in code line 111, you can see "timeout " is not a bypass msg.
 this means hermes is submitting transaction with mixing bypass and non-bypass msgs. This is just a test case we want to test.